test/ao/sparsity/test_sparsifier.py

```diff
@@ -55,3 +70,37 @@ def test_state_dict(self):
         assert sparsifier0.module_groups != sparsifier1.module_groups
         sparsifier1.load_state_dict(state_dict)
         assert sparsifier0.module_groups == sparsifier1.module_groups
+
+    def test_mask_squash(self):
+        model = Model()
+        sparsifier = ImplementedSparsifier(test=3)
+        sparsifier.prepare(model, [model.linear])
+        assert hasattr(model.linear, 'mask')
+        assert is_parametrized(model.linear, 'weight')
+        assert not hasattr(model.seq[0], 'mask')
+        assert not is_parametrized(model.seq[0], 'weight')
+
+        sparsifier.squash_mask()
+        assert not hasattr(model.seq[0], 'mask')
+        assert not is_parametrized(model.seq[0], 'weight')
+        assert not hasattr(model.linear, 'mask')
+        assert not is_parametrized(model.linear, 'weight')
+
+
+class TestWeightNormSparsifier(TestCase):
+    def test_constructor(self):
+        model = Model()
+        sparsifier = WeightNormSparsifier()
+        sparsifier.prepare(model, config=None)
+        for g in sparsifier.module_groups:
+            assert isinstance(g['module'], nn.Linear)
+            # The module_groups are unordered
+            assert g['path'] in ('seq.0', 'linear')
+
+    def test_logic(self):
+        model = Model()
+        sparsifier = WeightNormSparsifier(sparsity_level=0.5)
+        sparsifier.prepare(model, config=[model.linear])
+        sparsifier.enable_mask_update = True
+        sparsifier.step()
+        self.assertAlmostEqual(model.linear.mask.mean().item(), 0.5, places=2)
```

Comment:
Do we need a test to check that the masks are serialized correctly as part of the model?

Reply:
I thought those are tested when I test the parametrization state tests.